src/doc/rustc/src/codegen-options/index.md

```diff
@@ -172,6 +172,25 @@ values:
 
 The default if not specified depends on the target.
 
+## hint-mostly-unused
+
+This flag hints to the compiler that most of the crate will probably go unused.
+The compiler can optimize its operation based on this assumption, in order to
+compile faster. This is a hint, and does not guarantee any particular behavior.
+
+This option can substantially speed up compilation if applied to a large
+dependency where the majority of the dependency does not get used. This flag
```

Comment:
Can we provide any indication of how to do this? I've suggested some people try adding `-Zcross-crate-inline-threshold=always` to just one dependency and nobody seems to know how to do this offhand. If we're going to suggest people do a thing that's poorly understood I'd like to at least have a link to the Cargo docs about profile overrides.

Reply:
Sure, will do.

Reply:
I added documentation explaining how to do this with Cargo.